### PR TITLE
swagger-codegen: drop deprecated java_home_cmd

### LINKED ISSUE #61252: swagger-codegen install fails - MethodDeprecatedError java_home_cmd
# Bug report

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
  https://gist.github.com/ca0d050723941d81c59438364f8a4631
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

Install `swagger-codegen` via brew.

## What happened (include command output)

<details>
  <summary>Command output</summary>
  <pre>
brew install swagger-codegen
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 34 formulae.

==> Downloading https://github.com/swagger-api/swagger-codegen/archive/v3.0.20.tar.gz
==> Downloading from https://codeload.github.com/swagger-api/swagger-codegen/tar.gz/v3.0.20
######################################################################## 100.0%
Error: An exception occurred within a child process:
  MethodDeprecatedError: Calling Language::Java.java_home_cmd is disabled! Use Language::Java.java_home or Language::Java.overridable_java_home_env instead.
Please report this issue to the homebrew/core tap (not Homebrew/brew or Homebrew/core), or even better, submit a PR to fix it:
  /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/swagger-codegen.rb:21
  </pre>
</details>

## What you expected to happen

Successful install

## Step-by-step reproduction instructions (by running `brew install` commands)

`brew install swagger-codegen`

### PR DESCRIPTION
**Pull request by black-snow**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Closes https://github.com/Homebrew/homebrew-core/issues/61252  / old PR @linuxbrew: https://github.com/Homebrew/linuxbrew-core/pull/21113/commits/fa626a143fd7506f08178753326ad68428308656

### CODE REVIEW

File: Formula/swagger-codegen.rb [hunk 1]
```diff
@@ -18,8 +18,7 @@ class SwaggerCodegen < Formula
 
   def install
     # Need to set JAVA_HOME manually since maven overrides 1.8 with 1.7+
-    cmd = Language::Java.java_home_cmd("1.8")
-    ENV["JAVA_HOME"] = Utils.safe_popen_read(cmd).chomp
+    ENV["JAVA_HOME"] = Language::Java.java_home("1.8")
```
Comment (SMillerDev) [inline Formula/swagger-codegen.rb:21]: The github page mentions that it needs to be at least Java 8, so we're better of making the formula depend on `openjdk` and using `ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix` here.

Comment (black-snow) [reply Formula/swagger-codegen.rb:21]: thanks @SMillerDev 
I guess that would give the install location of openjdk - given that it's installed?
Is it a good practice to depend on other formulae? The formula depends on java 1.8 - is it enough to say `openjdk`? Shouldn't we restrict the version? Will `java` always be the same as `openjdk`? And why the hell do the tests fail? :|

Comment (SMillerDev) [reply Formula/swagger-codegen.rb:21]: - I guess that would give the install location of openjdk - given that it's installed?
That's what `ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix` will do

- Is it a good practice to depend on other formulae?
It's an excellent practice and the basis of homebrew. There's very few formula that don't do this.

- The formula depends on java 1.8 - is it enough to say openjdk?
You can replace the line `depends_on java` with `depends_on "openjdk"`

- Shouldn't we restrict the version?
No, homebrew always strives to use the latest dependencies. Restricting versions would make that more difficult.

- Will java always be the same as openjdk?
Yes, OpenJDK is an implementation of Java (and one that homebrew ships as a formula)

- And why the hell do the tests fail? :|
Gonna restart them to try again

Comment (SMillerDev) [reply Formula/swagger-codegen.rb:21]: Actually, if you push those changes it should restart again.

Comment (black-snow) [reply Formula/swagger-codegen.rb]: Aight, let's see if the tests run fine now.

File: Formula/swagger-codegen.rb [hunk 2]
```diff
@@ -14,11 +14,11 @@ class SwaggerCodegen < Formula
   end
 
   depends_on "maven" => :build
-  depends_on java: "1.8"
+  depends_on "openjdk": "1.8"
```
Comment (SMillerDev) [inline Formula/swagger-codegen.rb:17]:
```suggestion
  depends_on "openjdk"
```

**Comment by SMillerDev** [comment]
Can you squash the commits into one matching the PR title please

**Comment by black-snow** [comment]
> 
> 
> Can you squash the commits into one matching the PR title please

Is there some way to do this in the github GUI or only locally? You should have an option to [squash & merge](https://github.blog/2016-04-01-squash-your-commits/) with a single click.

**Comment by black-snow** [comment]
woops, missed one

**Comment by black-snow** [comment]
@SMillerDev should be fine now

**Review by SMillerDev — APPROVED** [review]

**Comment by SMillerDev** [comment]
Thanks @black-snow  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**